src/sentry/sentry_metrics/indexer/limiters/cardinality.py

```diff
@@ -99,22 +108,31 @@ def check_cardinality_limits(
                 ),
                 16,
             )
-            prefix = _build_quota_key(self.namespace, org_id)
+            prefix = _build_quota_key(message["use_case_id"], org_id)
             hash_to_offset[prefix, message_hash] = key
-            request_hashes[prefix].add(message_hash)
+            configured_quota = _construct_quotas(message["use_case_id"])
+
+            # since we might have some use cases that are covered by
+            # a quota and some that are not, only add entries that
+            # are covered by a quota
+            if configured_quota is not None:
+                request_hashes[(prefix, configured_quota)].add(message_hash)
```

Comment:
if there is only one quota configuration per usecase, I would change the mapping from `(prefix, quota) -> hashes` to `prefix -> (quota, hashes)`. or perhaps maintain a separate dictionary for quotas `prefix -> quota`

Reply:
maintaining a separate dict